examples/darksage.py

```diff
@@ -3569,88 +3569,93 @@ def jH2(self, tree, j_bin):
     def RadiusHI(self, tree, h):
         DiscRadii = np.zeros((len(tree),30))
         SigmaHI = np.zeros((len(tree),30))
+        arr = np.zeros(len(tree))
         for i in range(1,31):
             DiscRadii[:,i-1] = (tree['DiscRadii_'+str(i)]+tree['DiscRadii_'+str(i-1)])*0.5
             SigmaHI[:,i-1] = tree['DiscHI_'+str(i)]/(np.pi*(tree['DiscRadii_'+str(i)]**2-tree['DiscRadii_'+str(i-1)]**2))*1e-2*h
         (row, col) = np.where(SigmaHI>1.0)
         filt = np.append(np.diff(row)>0, True)
-        row, col = row[filt], col[filt]
-        arr = np.zeros(len(tree))
-        arr[row] = DiscRadii[row,col]
-        row, col = row[col<29], col[col<29]
-        arr[row] = arr[row] + (SigmaHI[row,col]-1.0)/(SigmaHI[row,col]-SigmaHI[row,col+1]) * (DiscRadii[row,col+1]-DiscRadii[row,col])
+        if len(filt[filt])>0:
```

Comment:
Not sure I understand - couldn't this simply be `len(filt) > 0`?

Reply:
I want to know if there is at least one True in the boolean 'filt' array.  If it were just len(filt) is would tell me the full length of the array, regardless of whether all entries were False or not.  So it's how I intended.

Reply:
Okay - thanks! 

An equivalent method is to check if *any* element is `True` -- and would corresponds to `if filt.any()`

Reply:
Also, there are lines in places that already discriminate between versions.  I've coded for self.Convertor.args.dataset_version to be 2016 or 2018 in places.  This was following our discussions last year.  I haven't changed the Converter class accordingly though, so let me do that...

Reply:
Actually, the "dataset_version" field is already defined at a higher level of the code.  Is this sufficient as is?  I.e. can one say feed --dataset_version when running the code and set to either 2016 or 2018?

Reply:
Yeah I was thinking about that - currently the version stored in there is the "Date of Conversion" (if nothing is passed). You could use the `model-name`, and switch case the various DarkSage releases (e.g., 2016, 2018) and then throw an error if there is no "year" in the model-name